indexd/index/drivers/alchemy.py

```diff
@@ -37,6 +40,94 @@
+        for permission in permissions:
+            method = permission['method']
+            service = permission['service']
+            if requested_method == "READ":
+                if "read-storage" == method or service == "indexd":
+                    authorized_resources.append(resource)
+            else:
+                if method in ["create", "update", "delete"]:
+                    authorized_resources.append(resource)
+    return authorized_resources
+
+
+def _check_user_access(authorized_resources: list[str], authz: list[str]) -> None:
+    """Raise an AuthzError if the user is not authorized for the given resources."""
+    for resource in authz:
+        if resource not in authorized_resources:
+            raise AuthzError(
+                f"User is not authorized for project {resource} {authz} {type(authz)}"
+            )
+
+
+def _is_rbac() -> bool:
+    """
+        Check if RBAC is enabled for the current app.
+        Does current user have access to a global discovery resource?
+        If so, RBAC is not enabled.
+    """
+    # For simplicity, we check the config directly.
+    are_records_discoverable = flask.current_app.config.get('ARE_RECORDS_DISCOVERABLE', True)
+    # If records are not discoverable, RBAC is not enabled.
+    if are_records_discoverable:
+        return False
+    # Does user have access to "GLOBAL_DISCOVERY_AUTHZ" resource?
+    global_discovery_authz: list = flask.current_app.config.get('GLOBAL_DISCOVERY_AUTHZ', [])
+    authorized_resources: list = _get_authorized_resources()
+    # if any of the global discovery authz resources are in the authorized resources, RBAC is not enabled
+    if any(resource in authorized_resources for resource in global_discovery_authz):
+        return False
+    # If we reach here, RBAC is enabled.
+    return True
+
+
+def _enforce_rbac(authz, method="READ") -> tuple[list[str], list[str]]:
+    """ Enforce RBAC for the current request."""
+    any_authz = None
+    if _is_rbac():
+        # If RBAC is not enabled, we don't need to check user access.
+        # This is for backwards compatibility with existing code.
+        authorized_resources = _get_authorized_resources(method)
+        if len(authorized_resources) == 0:
+            # Force filtering by an non-existent resource if no resources are authorized.
+            authorized_resources = ["NO-AUTHORIZED-RESOURCES"]
+        if authz:
+            _check_user_access(authorized_resources, authz or [])
+            any_authz = None
+        else:
+            authz = None
+            any_authz = authorized_resources
+
+    return any_authz, authz
+
+
+def _enforce_record_authz(record):
+    """ Enforce record authorization based on the current request's RBAC settings."""
+    if not _is_rbac():
+        return
+    authorized_resources = _get_authorized_resources()
+    if len(authorized_resources) == 0:
+        raise AuthError("User is not authorized for any resources")
```

Comment:
see other comments, leave this vague, just say user is not authorized

Reply:
done